### ISSUE TITLE
import pyarrow causes TypeError: an integer is required (got type bytes)

### ISSUE THREAD

**Issue by matanPercepto** [apache/arrow#37693]
### Describe the bug, including details regarding any error messages, version, and platform.

Hi,
I'm trying to use pyarrow, I installed it using pip, on python3.8, and installed version 12.0.1.
When I'm trying to import it in my code, during runtime I recieve the following error:
`File "/home/ubuntu/torch2/lib/python3.8/site-packages/pyarrow/__init__.py", line 65, in <module>
    import pyarrow.lib as _lib
  File "pyarrow/compat.pxi", line 43, in init pyarrow.lib
  File "/home/ubuntu/torch2/lib/python3.8/site-packages/cloudpickle/__init__.py", line 3, in <module>
    from cloudpickle.cloudpickle import *
  File "/home/ubuntu/torch2/lib/python3.8/site-packages/cloudpickle/cloudpickle.py", line 145, in <module>
    _cell_set_template_code = _make_cell_set_template_code()
  File "/home/ubuntu/torch2/lib/python3.8/site-packages/cloudpickle/cloudpickle.py", line 126, in _make_cell_set_template_code
    return types.CodeType(
TypeError: an integer is required (got type bytes)
`
I tried uninstalling and reinstalling, but it still happens.
Hope for your help.
Thanks

### Component(s)

Python

**Comment by AlenkaF** [comment]
This is probably a miss, but still, what is the version of cloudpickle?

**Comment by matanPercepto** [comment]
cloudpickle==0.4.0
is this a requirement as well?

**Comment by AlenkaF** [comment]
No, but it does find it in your env and so it is using it. Maybe it is causing issues due to being an old version (`2.0.0` is the latest version of cloudpickle).
Could you try running a different environment or upgrade cloudpickle?

**Comment by matanPercepto** [comment]
I updated cloudpickle to version 2.0.0 and it worked.
Thanks!
